programs/perpetuals/src/instructions/set_permissionless_oracle_pubkey.rs

```diff
@@ -0,0 +1,77 @@
+//! SetPermissionlessOraclePubkey instruction handler
+
+use {
+    crate::state::{
+        custody::Custody,
+        multisig::{AdminInstruction, Multisig},
+        perpetuals::Perpetuals,
+        pool::Pool,
+    },
+    anchor_lang::prelude::*,
+};
+
+#[derive(Accounts)]
+pub struct SetPermissionlessOraclePubkey<'info> {
+    #[account(mut)]
+    pub admin: Signer<'info>,
+
+    #[account(
+        mut,
+        seeds = [b"multisig"],
+        bump = multisig.load()?.bump
+    )]
+    pub multisig: AccountLoader<'info, Multisig>,
+
+    #[account(
+        seeds = [b"perpetuals"],
+        bump = perpetuals.perpetuals_bump
+    )]
+    pub perpetuals: Box<Account<'info, Perpetuals>>,
+
+    #[account(
+        seeds = [b"pool",
+                 pool.name.as_bytes()],
+        bump = pool.bump
+    )]
+    pub pool: Box<Account<'info, Pool>>,
+
+    #[account(
+        mut,
+        seeds = [b"custody",
+                 pool.key().as_ref(),
+                 custody.mint.as_ref()],
+        bump = custody.bump
+    )]
+    pub custody: Box<Account<'info, Custody>>,
+
+    system_program: Program<'info, System>,
```

Comment:
similar to above, removed this `system_program`. https://github.com/solana-labs/perpetuals/pull/33/files#r1357291887